sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParser.java

```diff
@@ -108,17 +120,17 @@ private void handleSequencePointTag(XmlParserHelper xmlParserHelper) {
       }
 
       if (files.containsKey(fileId)) {
-        String filePath = files.get(fileId);
+        Optional<String> optionalPath = files.get(fileId);
 
-        if (fileService.isSupportedAbsolute(filePath)) {
-          LOG.trace("OpenCover parser: add hits for fileId '{}', filePath '{}', line '{}', visitCount '{}'",
-            fileId, filePath, line, visitCount);
+        if (optionalPath.isPresent()) {
+          String absoluteFilePath = optionalPath.get();
+          coverage.addHits(absoluteFilePath, line, visitCount);
 
-          coverage.addHits(filePath, line, visitCount);
+          LOG.trace("OpenCover parser: add hits for fileId '{}', filePath '{}', line '{}', visitCount '{}'",
+            fileId, absoluteFilePath, line, visitCount);
         } else {
-          LOG.debug("Skipping the fileId '{}', line '{}', vc '{}' because file '{}'" +
-              " is not indexed or does not have the supported language.",
-            fileId, line, visitCount, filePath);
+          LOG.debug("Skipping the fileId '{}', line '{}', visitCount '{}' because file is not indexed or does not have the supported language.",
```

Comment:
Can we read the full path from xml in order to be able to print it? 

I'm thinking that it can be helpful when analyzing the logs to see directly the file name. Otherwise we will have to open the report and search for fileId.

Reply:
This is a conscious change I did. We read file paths in a different function (`handleFileTag`). Previously, we were storing the path from the coverage entry (actually, the "canonicalPath") in the `files` dictionary (map). Now, I store `Optional<String>` instead of the path.

Why did I do it? To avoid doing IO operations _(get the canonical path, to the scanner, to `fileService`)_ for each coverage and branch coverage entry.

It's a trade off here. I'm fine also with changing the approach and do the following instead:
- storing the original path from the coverage entry in the `files` map _(we need this for the deterministic build path resolution)_
- for each (branch) coverage entry: get the canonical path, and then make the `fileService` call if necessary

For the scanner search, I imagine it keeps all indexed files in memory. However, whenever it would find a not-indexed file, it will go to the filesystem search.

WDYT? It's a tradeoff between performance (and for large codebases, I imagine it could be a hit) and logs usefulness.

Reply:
From what I've noticed so far is that from time to time we do have either questions or issues related to coverage which are sometimes hard to investigate. I think that both performance and logging details are important and I think that having them both would be great 😄 

The solution you suggested with a class which contains both the original path and the resolved one by the file service sounds very good.